common/logger/src/macros.rs

```diff
@@ -99,6 +99,34 @@ macro_rules! schema {
         )
     };
 
+    // Identifier Keys debug
+    (@ { $(,)* $($out:expr),* }, ?$($k:ident).+ = $val:expr, $($args:tt)*) => {
+        $crate::schema!(
+            @ { $($out),*, &$crate::KeyValue::new($crate::__log_stringify!($($k).+), $crate::Value::from_serde(&format!("{:?}", $val))) },
+            $($args)*
+        )
+    };
+
+    (@ { $(,)* $($out:expr),* }, ?$($k:ident).+ = $val:expr) => {
+        $crate::schema!(
+            @ { $($out),*, &$crate::KeyValue::new($crate::__log_stringify!($($k).+), $crate::Value::from_serde(&format!("{:?}", $val))) },
+        )
+    };
+
+    // Identifier Keys display
+    (@ { $(,)* $($out:expr),* }, !$($k:ident).+ = $val:expr, $($args:tt)*) => {
+        $crate::schema!(
+            @ { $($out),*, &$crate::KeyValue::new($crate::__log_stringify!($($k).+), $crate::Value::from_serde(&$val.to_string())) },
+            $($args)*
+        )
+    };
+
+    (@ { $(,)* $($out:expr),* }, !$($k:ident).+ = $val:expr) => {
+        $crate::schema!(
+            @ { $($out),*, &$crate::KeyValue::new($crate::__log_stringify!($($k).+), $crate::Value::from_serde(&$val.to_string())) },
+        )
+    };
+
     // Literal Keys
     (@ { $(,)* $($out:expr),* }, $k:literal = $val:expr, $($args:tt)*) => {
```

Comment:
We probably also want to handle the literal case

Reply:
Btw, what's the reason we support literals?

Reply:
I mean we don't have to (i don't know if we actually use them anywhere) but its for the case where you want to name a field with a symbol that's not part of an identifier. E.g. "@timestamp"